test/PeriodicPrizePool.test.js

```diff
@@ -0,0 +1,255 @@
+
+const { ethers } = require('./helpers/ethers')
+const { expect } = require('chai')
+const buidler = require('./helpers/buidler')
+const getIterable = require('./helpers/iterable')
+
+const toWei = ethers.utils.parseEther
+const toBytes = ethers.utils.toUtf8Bytes
+const EMPTY_STR = toBytes('')
+
+const debug = require('debug')('ptv3:PeriodicPrizePool.test')
+
+const FORWARDER = '0x5f48a3371df0F8077EC741Cc2eB31c84a4Ce332a'
+
+let overrides = { gasLimit: 20000000 }
+
+
+describe.only('PeriodicPrizePool contract', function() {
+  let wallet, wallet2
+
+  let registry, governor, rngService, prizePool, prizeStrategy, cToken
+
+  let ticket, ticketCredit, sponsorship, sponsorshipCredit
+
+  let prizePeriodSeconds = toWei('1000')
+
+  beforeEach(async () => {
+    [wallet, wallet2] = await buidler.ethers.getSigners()
+
+    debug(`using wallet ${wallet._address}`)
+
+    debug('deploying registry...')
+    registry = await deploy1820(wallet)
+
+    debug('deploying protocol governor...')
+    governor = await deployContract(wallet, MockGovernor, [], overrides)
+
+    debug('deploying rng service...')
+    rngService = await deployContract(wallet, RNGServiceMock, [], overrides)
+
+    debug('deploying prize strategy...')
+    prizeStrategy = await deployContract(wallet, MockPrizeStrategy, [], overrides)
+  
+    debug('mocking tokens...')
+    cToken = await deployMockContract(wallet, CTokenInterface.abi, overrides)
+    ticket = await deployMockContract(wallet, Ticket.abi, overrides)
+    ticketCredit = await deployMockContract(wallet, ControlledToken.abi, overrides)
+    sponsorship = await deployMockContract(wallet, ControlledToken.abi, overrides)
+    sponsorshipCredit = await deployMockContract(wallet, ControlledToken.abi, overrides)
+
+    // Common Mocks for Tokens
+    await cToken.mock.underlying.returns(cToken.address)
+
+    debug('deploying prizePool...')
+    prizePool = await deployContract(wallet, CompoundPeriodicPrizePoolHarness, [], overrides)
+
+    debug('initializing prizePool...')
+    await prizePool.initialize(
+      FORWARDER,
+      governor.address,
+      prizeStrategy.address,
+      rngService.address,
+      prizePeriodSeconds,
+      cToken.address
+    )
+    debug('setting prizePool tokens...')
+    await prizePool.setTokens(
+      ticket.address,
+      sponsorship.address,
+      ticketCredit.address,
+      sponsorshipCredit.address
+    )
+  })
+
+  describe('initialize()', () => {
+    it('should set the params', async () => {
+      expect(await prizePool.getTrustedForwarder()).to.equal(FORWARDER)
+      expect(await prizePool.governor()).to.equal(governor.address)
+      expect(await prizePool.prizeStrategy()).to.equal(prizeStrategy.address)
+      expect(await prizePool.rng()).to.equal(rngService.address)
+      expect(await prizePool.prizePeriodSeconds()).to.equal(prizePeriodSeconds)
+      expect(await prizePool.cToken()).to.equal(cToken.address)
+    })
+  })
+
+  describe('setTokens()', () => {
```

Comment:
Maybe just add a test to ensure this can't be called twice (for the pesky require conditional)

Reply:
done!